instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/web/RestTemplateBeanPostProcessor.java

```diff
@@ -27,20 +28,22 @@ public Object postProcessAfterInitialization(Object bean, String beanName) {
       return bean;
     }
 
-    RestTemplate restTemplate = (RestTemplate) bean;
-    ClientHttpRequestInterceptor interceptor =
-        SpringWebTelemetry.create(openTelemetryProvider.getObject()).newInterceptor();
-    addRestTemplateInterceptorIfNotPresent(restTemplate, interceptor);
-    return restTemplate;
+    return addRestTemplateInterceptorIfNotPresent(
+        (RestTemplate) bean, openTelemetryProvider.getObject());
   }
 
-  private static void addRestTemplateInterceptorIfNotPresent(
```

Comment:
The modification of this class does not seem related to the instrumentation of the RestTemplateBuilder?

Reply:
Oh, I see now that the method is also used in `SpringWebInstrumentationAutoConfiguration`. Perhaps put the method use by both instrumentation in another class, something like `RestTemplateInterceptor`?

Reply:
good idea - done